components/GeoApi.js

```diff
@@ -0,0 +1,77 @@
+import React, { useEffect, useState } from "react";
+import { resolve } from "styled-jsx/css";
+import style from "../styles/Home.module.css";
+
+function GeoApi(props) {
+  const { weatherForecast, setWeatherForecast } = props;
+  const [locationName, setLocationName] = useState("Linn, Kansas");
+  const [searchTerm, setSearchTerm] = useState("");
+
+  const handleChange = (e) => {
+    setSearchTerm(e.target.value);
+  };
+
+  async function fetchCoords() {
+    let long = 0;
+    let lat = 0;
+    let firstRes = await fetch(`http://api.positionstack.com/v1/forward?access_key=${process.env.NEXT_PUBLIC_GEOLOCATION_API_KEY}&query=${searchTerm}`);
+    let firstData = await firstRes.json();
+    let a = 0;
+    while (firstData.data[a].continent !== "North America") {
+      a++;
+    }
+    long = firstData.data[a].longitude;
+    lat = firstData.data[a].latitude;
+    console.log(a, firstData);
```

Comment:
I believe the long and lat need to be returned here right? Or set the state?

Reply:
I was getting a lot of crashes so I separated fetch calls into 3 separate functions. This `fetchCoords()` is one of those functions. I'm planning on deleting the separate functions once I figure out all of the bugs.  I'm hoping to just use comments to explain all the different steps.

Reply:
Ah good thinking to separate into different functions! In fact I think the code would be overall easier to understand as separate and smaller functions. In general my preference is to have code that describes itself over requiring comments. One way to write descriptive code is to break blocks of code into functions and name the function in a way that describes the block of code. So what you did here was great, `fetchCoords` makes an API call with fetch to get coordinates!

Reply:
@12ahnaf34 what do you think of this?

Reply:
I agree, that would make the code easier to understand. I'll separate the code into smaller functions.